virttest/qemu_devices/qcontainer.py

```diff
@@ -2935,22 +2934,41 @@ def dimm_device_define_by_params(self, params, name):
 
     def memory_define_by_params(self, params, name):
         """
-        Create memory modules by params, include memory object and
-        pc-dimm devices.
+        Create memory modules by params, include memory object,
+        pc-dimm and virtio-mem devices.
         """
         params = params.object_params(name)
         devices = []
         if not self.has_device("pc-dimm"):
             LOG.warn("'PC-DIMM' does not support by your qemu")
             return devices
+        if not self.has_device("virtio-mem"):
+            LOG.warn("'virtio-mem' does not support by your qemu")
+            return devices
         mem = self.memory_object_define_by_params(params, name)
         if mem:
             devices.append(mem)
             use_mem = params.object_params(name).get("use_mem", "yes")
+            use_virtio_mem = params.object_params(name).get("use_virtio_mem", "no")
             if use_mem == "yes":
                 dimm = self.dimm_device_define_by_params(params, name)
                 dimm.set_param("memdev", mem.get_qid())
                 devices.append(dimm)
+            if use_virtio_mem == "yes":
+                virtio_mem_params = Params()
+                suffix = "_virtio_mem"
```

Comment:
Keep my opinion, @luckyh @yanan-fu, you can give your point, if we want to align to `_dimm`?

Reply:
That's a good point. My thought is we can consider to give a common suffix for all types of memory devices (e.g. `_memory` or `_memdev`), so that the related params being parsed in an unified way. And this time we can only apply it for virtio-mem, and leave the modification for dimm devices to the next stage.

Reply:
@luckyh that's a good point! I can apply whatever (`_memory` or `_memdev`) if you all agree @yanan-fu @PaulYuuu :blush:

Reply:
Yes, I prefer to pick `_memdev` as the suffix. But you can check if the usage exists.

Reply:
@luckyh @PaulYuuu 
Actually `memdev` is in use on tp-qemu to make reference to the property `memdev=` from the numa node object, so this could lead to a conflict...

Reply:
OK, thank you. So let's have a try of `_memory`, it makes sense in `memory_define_by_params`

Reply:
Great!